### PR TITLE
[ISSUE] 1463

### PR DESCRIPTION
**Pull request by IanCao**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

XXXXX

## Brief changelog

XX

## Verifying this change

add retry notify 4 times

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Comment by IanCao** [comment]
#1463

**Comment by zhaoyuguang** [comment]
From CI, I see the `Subscribe_ITCase.java` make the problem.
image
remove `import org.omg.CORBA.TIMEOUT` can fix it 
more detail see: https://github.com/alibaba/nacos/pull/1487

**Comment by IanCao** [comment]
> From CI, I see the `Subscribe_ITCase.java` make the problem.
> image
> 
> image
> remove `import org.omg.CORBA.TIMEOUT` can fix it
> more detail see: #1487

good job

File: client/src/main/java/com/alibaba/nacos/client/naming/core/EventDispatcher.java [hunk 1]
```diff
@@ -119,16 +119,15 @@ public void run() {
...
+                }
                 try {
-                    List<EventListener> listeners = observerMap.get(serviceInfo.getKey());
-
-                    if (!CollectionUtils.isEmpty(listeners)) {
-                        for (EventListener listener : listeners) {
-                            List<Instance> hosts = Collections.unmodifiableList(serviceInfo.getHosts());
-                            listener.onEvent(new NamingEvent(serviceInfo.getName(), serviceInfo.getGroupName(), serviceInfo.getClusters(), hosts));
-                        }
+                    for (EventListener listener : listeners) {
```
Comment (KomachiSion) [inline client/src/main/java/com/alibaba/nacos/client/naming/core/EventDispatcher.java:127]: I think that forEach in try code block can't solve the problem.

**Comment by KomachiSion** [comment]
PR is too old and no response in a long time. I will close this PR. If you want to contribute again, welcome open a new PR.

**Pull request closed without merging**